example/generator/odp_generator.c

```diff
@@ -1210,10 +1213,16 @@ int main(int argc, char *argv[])
 	/* Configure scheduler */
 	odp_schedule_config(NULL);
 
+	pkt_len = args->appl.payload + MAX_HDR_LEN;
+	if (pkt_len < POOL_PKT_LEN)
+		pkt_len = POOL_PKT_LEN;
+
+	printf("Packet length:      %u\n", pkt_len);
+
 	/* Create packet pool */
 	odp_pool_param_init(&params);
 	params.pkt.seg_len = POOL_PKT_LEN;
-	params.pkt.len     = POOL_PKT_LEN;
+	params.pkt.len     = pkt_len;
```

Comment:
Now packet len can exceed segment length, but setup_icmp_pkt() assumes that the whole packet data is contiguous in memory.

Reply:
setup_icmp_pkt always sends with payload of 56 bytes, so it will not an issue. Please correct me if I am wrong.

Reply:
@JannePeltonen any comments

Reply:
If someone changes setup_icmp_pkt_ref() to send longer ICMP packets and/or changes the segment size (POOL_PKT_LEN) to small enough, then setup_icmp_pkt() will overwrite the packet data buffer. This rather fragile since the code makes an assumption on what other pieces of code far way are doing. It would be better to check that the segment is large enough (e.g. by calling odp_packet_data_seg_len() instead of just odp_packet_data() and then checking the segment size against the requirement (headers + payload). Or by using a suitable static assert after defining the ICMP payload length as a macro).

Also, I do not think POOL_PKT_LEN is a good name anymore. I think it probably should be called POO_SEG_LEN or something similar (and the comment changed accordingly).

It would also be good to check with ODP_STATIC_ASSERT that SEG_LEN >= MAX_HDR_LEN as that is another assumption made in the code.